src/Common/CoreCommunications/ParsedTransactionMapper.cs

```diff
@@ -165,7 +165,7 @@ CancellationToken token
         }
 
         // We always use the top of the known ledger for this, as any estimates should be as present as possible
-        var stateVersionToUse = (await _dbContext.GetTopLedgerTransaction().SingleAsync(token)).ResultantStateVersion;
+        var stateVersionToUse = (await _dbContext.GetTopLedgerTransaction().FirstAsync(token)).ResultantStateVersion;
```

Comment:
If we're using `Take(1)` then I'm not sure this change is required?

Reply:
I guess it makes it more efficient by not returning 2, so that's probably worthwile :D